.github/workflows/create-release.yml

```diff
@@ -22,55 +22,120 @@ on:
       repo_owner: ${{ steps.repo.outputs.repo_owner }}
+      default_stack_dir: ${{ steps.lookup.outputs.default_stack_dir }}
+      stack_files_dir: ${{ steps.get-stacks.outputs.stack_files_dir }}
     steps:
       - name: Checkout repo
         uses: actions/checkout@v4
+        with:
+          fetch-depth: 0  # gets full history
 
-      - name: Get stack names and USN support
-        id: get-stack-names
+      - name: Get stacks images
+        id: get-stacks
         run: |
-          stacks=$(
+
+          if [[ -f ${{ env.STACKS_FILEPATH }} ]]; then
+            current_stacks=$(jq '[.images[] |
+            . +
+            {
+              "create_build_image": (.create_build_image // false),
+              "is_new": true
+            }]' ${{ env.STACKS_FILEPATH }} )
+
+            stack_files_dir="$(dirname "${{ env.STACKS_FILEPATH }}")"
+
+            git show $(git describe --tags --abbrev=0):"${{ env.STACKS_FILEPATH }}" > ./previous_images.json
+
+            if [ -f "./previous_images.json" ]; then
+              previous_stacks=$(jq '[.images[] |
+              . +
+              {
+                "create_build_image": (.create_build_image // false),
+                "is_new": false
+              }]' "./previous_images.json" )
+            else
+              previous_stacks="[]"
+            fi
+
+            stacks=$(echo "$current_stacks" | jq -c --argjson prev "$previous_stacks" '
+                map(
+                    . as $curr |
+                    ($prev[] | select(.name == $curr.name)) // $curr
+                )
+            ')
+          else
+            stacks=$(
             cat <<EOF
           [
             {
-              "name": "stack",
+              "name": "default",
               "config_dir": "stack",
               "output_dir": "build",
               "build_image": "build",
+              "build_receipt_filename": "build-receipt.cyclonedx.json",
               "run_image": "run",
-              "create_build_image": true
+              "run_receipt_filename": "run-receipt.cyclonedx.json",
+              "create_build_image": true,
+              "is_new": false
             }
           ]
           EOF
           )
+            stack_files_dir="stack"
+          fi
 
+          echo "stack_files_dir=$stack_files_dir" >> "$GITHUB_OUTPUT"
+
+          ## Filter stacks array to include the minimum number of attributes
+          stacks=$(echo "$stacks" | jq 'map({
+              name,
+              config_dir,
+              output_dir,
+              build_image,
+              build_receipt_filename,
+              run_image,
+              run_receipt_filename,
+              create_build_image,
+              base_build_container_image,
+              base_run_container_image,
+              is_new
+            })')
+
+          stacks=$(jq -c <<< "$stacks" )
+          printf "stacks=%s\n" "${stacks}" >> "$GITHUB_OUTPUT"
+
```

Comment:
Discussed ,add comment here explaining the two models and that this section figures out which model the stack fits into.